examples/README.rst

```diff
@@ -13,3 +13,14 @@ series (magnitude and phase images) and respiratory trace information obtained f
 multi-gradient echo (MGRE) magnitude image is used to generate an ROI and resample the static and dynamic Gx, Gy, Gz
 component maps to match the MGRE image. Lastly the average Gx, Gy, Gz values within the ROI are computed for each
 slice.
+
+demo_b1_shimming.sh
+--------------------------
+
+This  `shell script <https://github.com/shimming-toolbox/shimming-toolbox/blob/master/examples/demo_b1_shimming.sh>`__
```

Comment:
link to script should be relative so it works for branches

Reply:
I tried to use relative paths in a .rst file but it did not work... Maybe I missed something but from what I understand, Sphinx builds the doc from the GitHub repo and puts it under the `readthedocs.org` server. So by using relative paths, one can only point to html pages that have been added to this RTD server and not to files located on GitHub... 
Maybe a solution would be to fetch the branch name during the documentation build and insert it in the .rst file links. 
@kousu @joshuacwnewton do you have any thoughts ?

Reply:
A couple of notes here:

* The following syntax:
   ```   
   ` <>`_
   ```
   Is meant for [external hyperlinks only](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#hyperlinks), i.e. weblinks to be opened in the browser. So, directly changing this to a relative link wouldn't work as-is -- we would need to change the syntax, too.
* But, the internal link syntax (e.g. [`:ref:`](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#internal-links)) won't work either, for the reasons that @gaspardcereza highlighted -- the internal links are meant to point to documentation pages (i.e. reST/`.rst` files) that get converted to html files.
* We could still use the [`:download:`](https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#role-download) role! But, in that case, the functionality would be different (clicking the link would download the script, rather than opening a page on GitHub).
* So, it depends on the purpose of the link... 

But, why do we want the link to work for branches anyway? Isn't this link only going to be viewed by the user on master? (In which case, I would think the original solution is probably OK.)

Reply:
The idea behind having the links pointing to the branch was mainly for review purpose but once merged it does not matter anymore so I think the `:download:` solution would be a bit overkill here